e2e/cypress/e2e/dynamic-groups/dynamic-groups.cy.ts

```diff
@@ -207,34 +199,47 @@ describe("dynamic groups", () => {
       ]);
     });
 
-    context("unordered", () => {
-      it("should create dynamic groups based on scene_id", () => {
-        cy.get("[data-cy=entry-counts").should("have.text", "200 groups");
+    ["left", "right", "pcd"].forEach((slice) => {
+      context(`unordered with slice = ${slice}`, () => {
+        before(() => {
+          cy.waitForGridToBeVisible("quickstart-groups-dynamic");
+          cy.clearViewStages();
 
-        // todo: investigate, "first()" is used to suppress this flakiness;
-        // sometimes an extra element is rendered and the test fails
-        cy.get("[data-cy=dynamic-group-pill-button]").click();
-        cy.get("[data-cy=order-by-selector]").should("not.exist");
-        cy.get("[data-cy=group-by-selector]").click();
-        cy.get("[data-cy=selector-result-scene_id]").click();
+          cy.get("[data-cy=entry-counts").should("have.text", "200 groups");
+          // todo: investigate, "first()" is used to suppress this flakiness;
+          // sometimes an extra element is rendered and the test fails
+          cy.get("[data-cy=dynamic-group-pill-button]").click();
+          cy.get("[data-cy=order-by-selector]").should("not.exist");
+          cy.get("[data-cy=group-by-selector]").click();
+          cy.get("[data-cy=selector-result-scene_id]").click();
 
-        cy.get("[data-cy=dynamic-group-btn-submit]").click();
+          cy.get("[data-cy=dynamic-group-btn-submit]").click();
 
-        cy.get("[data-cy=entry-counts").should("have.text", "8 groups");
+          cy.get("[data-cy=entry-counts").should("have.text", "8 groups");
 
-        cy.get("[data-cy=checkbox-scene_id]").click();
-        cy.get("[data-cy=checkbox-timestamp]").click();
+          cy.get("[data-cy=checkbox-scene_id]").click();
+          cy.get("[data-cy=checkbox-timestamp]").click();
 
-        cy.get("[data-cy=looker]").should("have.length", 8);
+          cy.get("[data-cy=looker]").should("have.length", 8);
+        });
 
-        cy.get("[data-cy=looker]").first().click();
+        beforeEach(() => {
+          cy.get("[data-cy=selector-slice]").first().click();
```

Comment:
I think `POM` wrappers would make code like this a lot easier to understand. Right now I'm not sure what happens before each test. For now a comment would be enough